packages/esm-patient-banner-app/src/print-identifier-sticker/print-identifier-sticker.modal.tsx

```diff
@@ -0,0 +1,151 @@
+      subtitle:
+        getCoreTranslation('printErrorExplainer', 'An error occurred in "{{errorLocation}}": ', { errorLocation }) +
+        error,
+    });
+
+    setIsPrinting(false);
+  }, []);
+
+  const handlePrint = useReactToPrint({
+    content: () => contentToPrintRef.current,
+    documentTitle: `${getPatientName(patient)} - ${headerTitle}`,
+    onAfterPrint: handleAfterPrint,
+    onBeforeGetContent: handleBeforeGetContent,
+    onPrintError: handlePrintError,
+    copyStyles: true,
+  });
+
+  return (
+    <>
+      <ModalHeader
+        closeModal={closeModal}
+        title={getCoreTranslation('printIdentifierSticker', 'Print identifier sticker')}
+      />
+      <ModalBody>
+        <div ref={contentToPrintRef}>
+          <style type="text/css" media="print">
+            {`
+              @page {
+                size: ${pageSize};
+              }
+            `}
+          </style>
+          <PrintComponent patient={patient} />
+        </div>
+      </ModalBody>
+      <ModalFooter>
+        <Button kind="secondary" onClick={closeModal}>
+          {getCoreTranslation('cancel', 'Cancel')}
+        </Button>
+        <Button className={styles.button} disabled={isPrinting} onClick={handlePrint} kind="primary">
+          {isPrinting ? (
+            <InlineLoading className={styles.loader} description={getCoreTranslation('printing', 'Printing') + '...'} />
+          ) : (
+            getCoreTranslation('print', 'Print')
+          )}
+        </Button>
+      </ModalFooter>
+    </>
+  );
+};
+
+interface PrintComponentProps extends Partial<ConfigObject> {
+  patient: fhir.Patient;
+}
+
+const PrintComponent = ({ patient }: PrintComponentProps) => {
+  const { t } = useTranslation();
+  const { printPatientSticker } = useConfig<ConfigObject>();
+  const primaryIdentifierValue = patient?.identifier?.find((identifier) => identifier.use === 'official')?.value;
+  return (
+    <div className={styles.stickerContainer}>
+      <div className={styles.documentHeader}>
+        {printPatientSticker?.header?.showBarcode && (
+          <Barcode value={primaryIdentifierValue} {...defaultBarcodeParams} />
+        )}
+        {printPatientSticker?.header?.showLogo && (
+          <div className={styles.implementationLogo}>
+            <ImplementationLogo />
+          </div>
+        )}
+      </div>
+      {printPatientSticker.fields.map((field) => {
+        const Component = getPatientField(field);
+        return <Component key={field} patient={patient} />;
+      })}
+    </div>
+  );
+};
+
+const ImplementationLogo: React.FC = () => {
+  const { t } = useTranslation();
+  const { printPatientSticker } = useConfig<ConfigObject>();
+
+  return printPatientSticker?.header?.logo ? (
+    <img alt="implementation-logo" src={printPatientSticker?.header?.logo} />
+  ) : (
+    <svg data-testid="openmrs-logo" role="img" viewBox="0 0 380 119" xmlns="http://www.w3.org/2000/svg">
```

Comment:
Replace this with an SVG sprite using `#omrs-logo-full-mono`

Reply:
I tried with the above sprite, but it's not rendering in the iframe created by the library when printing the sticker.

Reply:
I've left a TODO comment in the logic as a reminder to look into why this isn't working.